File: study-planner/scripts/extract_sb_orders.js

const fs = require('fs');
const pdf = require('pdf-parse');

const dataBuffer = fs.readFileSync('D:\\IP 2026\\Dak Sutra Source\\SB_Order_2019-25_English-07032026.pdf');

pdf(dataBuffer).then(function(data) {
    fs.writeFileSync('D:\\IP 2026\\study-planner\\scratch\\sb_orders_text.txt', data.text);
    console.log('PDF extracted successfully to sb_orders_text.txt');
}).catch(err => {
    console.error('Error reading PDF:', err);
});

File: study-planner/scripts/extract_sb_orders.py
from pypdf import PdfReader
import sys

pdf_path = r"D:\IP 2026\Dak Sutra Source\SB_Order_2019-25_English-07032026.pdf"
output_path = r"d:\IP 2026\study-planner\scratch\sb_orders_text.txt"

try:
    reader = PdfReader(pdf_path)
    text = ""
    for page in reader.pages:
        text += page.extract_text() + "\n"
    
    with open(output_path, "w", encoding="utf-8") as f:
        f.write(text)
    print(f"Successfully extracted text to {output_path}")
except Exception as e:
    print(f"Error: {e}")
    sys.exit(1)
